include/swift/AST/USRGeneration.h

```diff
@@ -47,8 +47,14 @@ bool printDeclTypeUSR(const ValueDecl *D, raw_ostream &OS);
 bool printValueDeclUSR(const ValueDecl *D, raw_ostream &OS);
 
 /// Prints out the USR for the given ModuleEntity.
+/// \param useRealNameIfAliased Whether to use the module's real name in case
+///                             module aliasing is used. For example, if a file
+///                             has `import Foo` and `-module-alias Foo=Bar` is
+///                             passed, treat Foo as an alias and Bar as the real
+///                             module name as its dependency. This only applies
+///                             to Swift modules.
 /// \returns true if it failed, false on success.
-bool printModuleUSR(ModuleEntity Mod, raw_ostream &OS);
+bool printModuleUSR(ModuleEntity Mod, raw_ostream &OS, bool useRealNameIfAliased = false);
```

Comment:
Is there ever a reason to make a USR using a non-canonical name?  I would have expected USRs to always use the real name unconditionally.  It's not good for us to have two USRs referring to the same thing.   If we really need this to be configurable, the default should probably be flipped, or else there should be no default value.